From: Jeff Layton <jlayton@redhat.com>
Date: Wed, 16 Dec 2009 19:07:50 -0500
Subject: [sunrpc] Don't display zero scope IDs
Message-id: <1260990470-32248-3-git-send-email-jlayton@redhat.com>
Patchwork-id: 22008
O-Subject: [RHEL6.0 PATCH 2/2] BZ#463530: SUNRPC: Don't display zero scope IDs
Bugzilla: 463530
RH-Acked-by: Peter Staubach <staubach@redhat.com>

From: Chuck Lever <chuck.lever@oracle.com>

A zero scope ID means that it wasn't set, so we don't need to append
it to presentation format addresses.

Signed-off-by: Chuck Lever <chuck.lever@oracle.com>
Acked-by: Jeff Layton <jlayton@redhat.com>

Signed-off-by: Aristeu Rozanski <arozansk@redhat.com>

diff --git a/net/sunrpc/addr.c b/net/sunrpc/addr.c
index 56febe3..008e36a 100644
--- a/net/sunrpc/addr.c
+++ b/net/sunrpc/addr.c
@@ -81,6 +81,8 @@ static size_t rpc_ntop6(const struct sockaddr *sap,
 
 	if (!(ipv6_addr_type(&sin6->sin6_addr) & IPV6_ADDR_LINKLOCAL))
 		return len;
+	if (sin6->sin6_scope_id == 0)
+		return len;
 
 	rc = snprintf(scopebuf, sizeof(scopebuf), "%c%u",
 			IPV6_SCOPE_DELIMITER, sin6->sin6_scope_id);
